python/tvm/tir/tensor_intrin/cuda.py

```diff
@@ -36,7 +36,7 @@ def shared_16x32_to_ldmatrix_32x16_layout(i, j):
 
 
 def shared_32x16_to_ldmatrix_32x16_layout(i, j):
-    thread_id = (i % 4) + 4 * (j % 8)
+    thread_id = (i % 16) // 4 + 4 * (j % 8)
```

Comment:
cc @masahi

Reply:
Thank you for tagging @masahi, I had forgotten to do so.  I *think* I have it set up correctly, based on Nvidia documentation and similarity to the (16,32) shape, but couldn't verify definitively.

Reply:
hmm I think the original mapping is correct, this is from p34 of the slide https://developer.download.nvidia.com/video/gputechconf/gtc/2020/presentations/s21745-developing-cuda-kernels-to-push-tensor-cores-to-the-absolute-limit-on-nvidia-a100.pdf

Sorry I don't remember the details

Reply:
ah sorry I was talking about `shared_16x32_to_ldmatrix_32x16_layout`. I need to remember how I came up with shared_32x16_to_ldmatrix_32x16_layout

Reply:
Even if the index map is incorrect, it doesn't affect the correctness of tensorized MMA since the index map is only used for pattern matching purpose...

Reply:
Thank you for looking into it!  I wasn't able to find any tests that explicitly validate the transform (e.g. use the transform to generate data in a specific layout, then pass through the mma), as all the tests either started with transformed data, only used the 16x16 shape, or replaced everything with the tensor intrinsic.

I had put together [this standalone test](https://gist.github.com/Lunderberg/0c2a44de34e7e2a1d149c37b2a112f91) to convince myself on it.  The main issue with the current index map is that it doesn't map to unique locations (512 input indices map to 128 output indices).  It only arose as an issue in this PR, because it generates the inverse in order to determine whether/where padding is required.